src/server/main.py

```diff
@@ -0,0 +1,382 @@
+        and is protected according to the licence of the said module :
+        BSD 3-Clause "New" or "Revised" License. Please see
+        https://github.com/pallets/flask/blob/main/LICENSE.rst
+        for more information.
+
+        .. admonition:: Keep in Mind
+           Flask will suppress any server error with a generic error page
+           unless it is in debug mode.  As such to enable just the
+           interactive debugger without the code reloading, you have to
+           invoke :meth:`run` with ``debug=True`` and ``use_reloader=False``.
+           Setting ``use_debugger`` to ``True`` without being in debug mode
+           won't catch any exceptions because there won't be any to
+           catch.
+        :param host: the hostname to listen on. Set this to ``'0.0.0.0'`` to
+            have the server available externally as well. Defaults to
+            ``'127.0.0.1'`` or the host in the ``SERVER_NAME`` config variable
+            if present.
+        :param port: the port of the webserver. Defaults to ``5000`` or the
+            port defined in the ``SERVER_NAME`` config variable if present.
+        :param debug: if given, enable or disable debug mode. See
+            :attr:`debug`.
+        :param load_dotenv: Load the nearest :file:`.env` and :file:`.flaskenv`
+            files to set environment variables. Will also change the working
+            directory to the directory containing the first file found.
+        :param options: the options to be forwarded to the underlying Werkzeug
+            server. See :func:`werkzeug.serving.run_simple` for more
+            information.
+        .. versionchanged:: 1.0
+            If installed, python-dotenv will be used to load environment
+            variables from :file:`.env` and :file:`.flaskenv` files.
+            The :envvar:`FLASK_DEBUG` environment variable will override :attr:`debug`.
+            Threaded mode is enabled by default.
+        .. versionchanged:: 0.10
+            The default port is now picked from the ``SERVER_NAME``
+            variable.
+
+        """
+        self.app.run(**kwargs)
+
+    def route_run(self):
+        """
+        Handle the start of an instance.
+
+        expected form :
+
+        {
+            "type":"drl|ga",
+            "params":{
+                "nb_epochs":"int",
+                "pop_size?":"int",
+                "mutation_rate?":"float",
+                "crossover_rate?":"float",
+                "learning_rate?":"float",
+                "batch_size?":"int",
+                "momentum?":"float"
+            },
+            "override":"bool",
+            "bench_id":"string",
+            "snapshot_rate":"int",
+            "seed":"int"
+        }
+
+        Returns :
+        {
+            “busy”:”bool”
+        }
+
+
+        """
+        if self._runner is not None:
+            return {"busy": True}
+
+        if request.method == "POST":
+            print(request.form)
```

Comment:
data = json.loads(request.data)["data"]
and you can continue with this....
example : metho = data["type"]

Reply:
Please add sentences to explain...